### PR TITLE
[docs] Improve AppAuth docs

### PR DESCRIPTION
**Pull request by ccheever**
- Explain OAuth a little bit more
- Mention that there are specialized modules for Google and Facebook
- Give a complete example as an inline Snack rather than just as
    sourc code

# Why

I wished that there were an inline Snack myself so I made one from the existing example.
And I added text to make it more clear to me.

# How

I typed out the code from the example given already and then added a simple UI for it.

# Test Plan

I tested the inline Snack I made on Android and iOS. I tested the links to the Facebook and Google sign in modules manually.

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 1]
```diff
@@ -3,10 +3,17 @@ title: AppAuth
...
+import SnackInline from '~/components/plugins/SnackInline';
 import TableOfContentSection from '~/components/plugins/TableOfContentSection';
 
 **`expo-app-auth`** allows you to authenticate and authorize your users through the native OAuth library AppAuth by [OpenID](https://github.com/openid).
 
+Many services that let you authenticate with them or login with them, like GitHub, Google, GitLab, etc., use the OAuth 2.0 protocol. It's the industry standard.
+
+If you are trying to implement sign in with [Google](../google-sign-in) or [Facebook](../facebook), there are special modules in the Expo SDK for those (though this module will work).
+
+For now, this module only works on Anroid and iOS, but web support is possible is planned to be added.
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
Currently, this module only supports on Android, and iOS. Web support is planned to be added. Track it here: [Web support in expo-app-auth](https://github.com/expo/expo/issues/6883).
```

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 2]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
 
 Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth Sign-In provider.
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth sign in provider.
```

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 3]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
...
+      if (cachedAuth && !authState) {
+        setAuthState(cachedAuth);
+      }
+    })();
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      <Text>Expo AppAuth Example</Text>
+      <Text></Text>
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion

```

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 4]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
...
+      <Button
+        title="Sign In with Google "
+        onPress={async () => {
+          let _authState = await signInAsync();
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
          const authState = await signInAsync();
```
no underscore

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 5]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
...
+        }}
+      />
+      <Button
+        title="Sign Out "
+        onPress={async () => {
+          await signOutAsync(authState);
+          setAuthState(null);
+        }}
+      />
+      <Text>{JSON.stringify(authState)}</Text>
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
      <Text>{JSON.stringify(authState, null, 2)}</Text>
```

human-readable format for the JSON.

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 6]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
...
+    </View>
+  );
+}
+
+let styles = StyleSheet.create({
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
const styles = StyleSheet.create({
```

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 7]
```diff
@@ -80,53 +113,33 @@ async function getCachedAuthAsync() {
...
- * When our session expires, we can exchange the refresh token to get new auth tokens.
- * > Auth tokens are not the same as a Refresh token
- *
- * Not every provider (very few actually) will return a new "Refresh Token".
- * This just means the user will have to Sign-In more often.
- */
 async function refreshAuthAsync({ refreshToken }) {
-  const authState = await AppAuth.refreshAsync(config, refreshToken);
-  console.log('refreshAuthAsync', authState);
+  let authState = await AppAuth.refreshAsync(config, refreshToken);
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]: why `let`?

File: docs/pages/versions/unversioned/sdk/app-auth.md [hunk 8]
```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
 
 Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth Sign-In provider.
 
+<SnackInline>
+
 ```js
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/app-auth.md]:
```suggestion
```tsx
```

better colors

**Review by EvanBacon — CHANGES_REQUESTED** [review]
added feedback

File: docs/pages/versions/unversioned/sdk/app-auth.md (hunk 6 above)
Comment (ccheever) [reply docs/pages/versions/unversioned/sdk/app-auth.md]: I think `const` is overused in most circumstances. 
https://overreacted.io/on-let-vs-const/

File: docs/pages/versions/unversioned/sdk/app-auth.md (hunk 7 above)
Comment (ccheever) [reply docs/pages/versions/unversioned/sdk/app-auth.md]: I just use `let` everywhere unless I want to strongly indicated intent to make something `const`.

Similar reasons as described here. https://overreacted.io/on-let-vs-const/

File: docs/pages/versions/unversioned/sdk/app-auth.md (hunk 8 above)
Comment (ccheever) [reply docs/pages/versions/unversioned/sdk/app-auth.md]: cool

File: docs/pages/versions/unversioned/sdk/app-auth.md (hunk 3 above)
Comment (ccheever) [reply docs/pages/versions/unversioned/sdk/app-auth.md]: I think I was just adding whitespace here but there's probably a better way to do it and this looks sloppy.

File: docs/pages/versions/unversioned/sdk/app-auth.md (hunk 4 above)
Comment (ccheever) [reply docs/pages/versions/unversioned/sdk/app-auth.md]: same as other comments about `let` and `const`

**Review by ccheever — COMMENTED** [review]
Am I doing this right?

**Review by ccheever — COMMENTED** [review]
Made suggested changes after talking with @EvanBacon

**Pull request merged**